__tests__/components/TransactionOptinPayoutDetails.spec.ts

```diff
@@ -0,0 +1,520 @@
+            },
+        };
+
+        return getComponent(
+            TransactionOptinPayoutDetails,
+            {
+                network: mockNetworkStore,
+            },
+            {},
+            props,
+            {
+                Spin: true,
+            },
+            dispatch,
+        );
+    };
+
+    describe('referenceInnerTransactions', () => {
+        const runBasicReferenceInnerTransactionsTests = (currentAccount, innerTransactions) => {
+            // Arrange:
+            const wrapper = getTransactionOptinPayoutDetailsWrapper({
+                currentAccount,
+                transaction: createMockAggregateTransaction(innerTransactions),
+            });
+
+            // Act + Assert:
+            // @ts-ignore
+            expect(wrapper.vm.referenceInnerTransactions).toEqual([]);
+        };
+
+        test('returns transaction when transfer transaction included mosaic', () => {
+            // Arrange:
+            const transferTransaction = createMockTransferTransaction([new Mosaic(new MosaicId('461A9811A1DC1FBB'), UInt64.fromUint(0))]);
+
+            const wrapper = getTransactionOptinPayoutDetailsWrapper({
+                currentAccount: WalletsModel1,
+                transaction: createMockAggregateTransaction([transferTransaction, createMockMultisigAccountModificationTransaction()]),
+            });
+
+            // Act:
+            // @ts-ignore
+            const transactions = wrapper.vm.referenceInnerTransactions;
+
+            // Assert:
+            expect(transactions.length).toBe(1);
+            expect(transactions[0]).toEqual(transferTransaction);
+        });
+
+        test('returns an empty array when transaction type is not transfer transaction', () => {
+            runBasicReferenceInnerTransactionsTests(WalletsModel1, [createMockMultisigAccountModificationTransaction()]);
+        });
+
+        test('returns an empty array when the transfer transaction does not include mosaic', () => {
+            runBasicReferenceInnerTransactionsTests(WalletsModel1, [createMockTransferTransaction()]);
+        });
+
+        test("returns an empty array when the transaction's recipient is not current account", () => {
+            runBasicReferenceInnerTransactionsTests(WalletsModel2, [createMockTransferTransaction()]);
+        });
+    });
+
+    describe('hasTransfers', () => {
+        const runBasicHasTransfersTests = (innerTransaction, expectedResult) => {
+            // Arrange:
+            const wrapper = getTransactionOptinPayoutDetailsWrapper({
+                currentAccount: WalletsModel1,
+                transaction: createMockAggregateTransaction(innerTransaction),
+            });
+
+            // Act + Assert:
+            // @ts-ignore
+            expect(wrapper.vm.hasTransfers).toBe(expectedResult);
+            expect(wrapper.find('table').exists()).toBe(expectedResult);
```

Comment:
consider merging check of hasTransfers with previous group, since the two properties are essentially linked